roles/openldap/defaults/main.yml

```diff
@@ -0,0 +1,52 @@
+keystone:
+  ldap:
+    enabled: False 
+
+    # ldap server settings
+    url: ldap://0.0.0.0:389
```

Comment:
Yes so couple of changes here. 
1) The url that keystone connects to should be in defaults of the keystone role
2) Provide a variable which specifies which urls openldap listens on, openldap will pickup interfaces and ports to listen on based on $SLAPD_SERVICES in /etc/default/slapd, the default will listen on the interface that has the ip which is backend for haproxy (same as the one on which keystone is listening to on the controller).